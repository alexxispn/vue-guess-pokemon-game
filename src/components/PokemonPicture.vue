<template>
  <div class="pokemon-img">
    <img src="../assets/img/blue.svg" class="blue fade-in" alt="background blue" width="300"/>
    <img v-if="!showPokemon" :src="imageUrl" class="pokemon hidden-pokemon" alt="pokemon">
    <img v-else :src="imageUrl" class="pokemon fade-in" alt="pokemon">
  </div>
</template>

<script>
export default {
  name: "PokemonPicture",
  props: {
    pokemonId: {
      type: Number,
      required: true
    },
    showPokemon: {
      type: Boolean,
      required: true,
      default: false
    }
  },
  computed: {
    imageUrl() {
      return `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${this.pokemonId}.svg`
    }
  }
}
</script>

<style scoped>

.pokemon-img{
  display: grid;
  grid-template-areas: "picture";
  place-items: center;
}

.blue {
  grid-area: picture;
  width: clamp(min(30vh, 30vw), 90%, max(50vh, 40vw));
  filter: brightness(80%);
}

.pokemon {
  grid-area: picture;
  width: clamp(min(20vh, 20vw), 50%, max(30vh, 25vw));
  z-index: 1;
}

.hidden-pokemon {
  filter: brightness(0);
}


</style>
