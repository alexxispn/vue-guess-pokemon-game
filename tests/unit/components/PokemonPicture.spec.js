import PokemonPicture from "@/components/PokemonPicture";
import {shallowMount} from "@vue/test-utils";

describe('PokemonPicture', () => {
    describe('when showPokemon:false', () => {
        const wrapper = shallowMount(PokemonPicture, {
            props: {
                pokemonId: 100,
                showPokemon: false
            }
        })
        it('should match with snapshot', () => {
            expect(wrapper.html()).toMatchSnapshot()
        })
        it('should return only 2 images, img2 with class hidden', () => {
            const [img1, img2, img3] = wrapper.findAll('img')
            expect(img1.exists()).toBeTruthy()
            expect(img2.exists()).toBeTruthy()
            expect(img3).toBe(undefined)
            expect(img2.classes('hidden-pokemon')).toBeTruthy()
        })
        it('should return imageUrl ended in 100.svg', () => {
            const [, img2] = wrapper.findAll('img')
            expect(img2.attributes('src')).toBe('https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg')
        })
    })
    describe('when showPokemon:true', () => {
        const wrapper = shallowMount(PokemonPicture, {
            props: {
                pokemonId: 100,
                showPokemon: true
            }
        })
        it('should return the second image without class hidden', () => {
            const [, img2] = wrapper.findAll('img')
            expect(img2.classes('hidden-pokemon')).toBeFalsy()
        })
    })
})
