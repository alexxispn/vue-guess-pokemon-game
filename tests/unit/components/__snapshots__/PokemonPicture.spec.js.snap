// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPicture when showPokemon:false should match with snapshot 1`] = `<div class="pokemon-img"><img src="" class="blue fade-in" alt="background blue" width="300"><img src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg" class="pokemon hidden-pokemon" alt="pokemon"></div>`;
